osquery/bro/utils.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <iostream>
+
+#include <boost/algorithm/string/replace.hpp>
+#include <boost/algorithm/string/trim.hpp>
+
+#include <broker/broker.hh>
+#include <broker/endpoint.hh>
+#include <broker/message_queue.hh>
+#include <broker/report.hh>
+
+#include <osquery/config.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+
+#include "osquery/bro/utils.h"
+#include "osquery/core/json.h"
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+
+Status createSubscriptionRequest(const std::string& rType,
+                                 const broker::message& msg,
+                                 const std::string& incoming_topic,
+                                 SubscriptionRequest& sr) {
+  // Check number of fields
+  unsigned long numFields;
+  if (rType == "EXECUTE")
```

Comment:
There's inconsistent use of `{` here, but we recommend adding explicit scope `{}` for 1-line conditionals, it helps reduce future bugs.